### DISCUSSION TITLE
Type binding mismatch: javax.security.Subject instead of my x.y.z.Subject

### DISCUSSION THREAD

**Discussion by agazzarini** [graphql-java-kickstart/graphql-spring-boot#490] [Q&A] [answered]
I have a schema which declare a type:

```
type Subject {
  uri:String!
}
```

and a class 

```
public class Subject {
   private String uri;

   // Getters and setters
}
```

At startup, instead of picking up my Subject class, I get the following error: 

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.kickstart.tools.SchemaParser]: Factory method 'schemaParser' threw exception; nested exception is graphql.kickstart.tools.resolver.FieldResolverError: No method or field found as defined in schema <unknown>:338 with any of the following signatures (with or without one of [interface graphql.schema.DataFetchingEnvironment] as the last argument), in priority order:
  javax.security.auth.Subject.uri()
  javax.security.auth.Subject.getUri()
  javax.security.auth.Subject.uri
```

I'm struggling, trying to find a way to indicate my class instead of that coming from javax.security. Apart asking where I can set that mapping, I think it would be better to have a configuration or something like that, which would allow me to indicate a package where the bindings can be found. I mean, it's absolutely great to rely on this automatic binding, but in these cases there should be an easy way to solve the ambiguity.

**Comment by oliemansm** [comment, marked as answer]
The binding is handled by graphql-java-tools by starting the scan of the root Query and Mutation types and following all defined queries and mutations to bind all types and inputs to the corresponding classes. The above snippet above doesn't show that crucial part.

This `Subject` type is used in one of your queries or mutations? And in the Java class implementing that query or mutation you are using your own Subject class? Because in that case it should map it correctly.

**Comment by agazzarini** [comment]
Thanks for your hints, @oliemansm. As expected, it was a fault of mine: I had one of my classes with a wrong import (javax.auth.Subject instead of my Subject). I think I've checked that class more or less 200 times but nothing, when you don't want to see a thing there's nothing to do.

Thanks again
Andrea
